compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -953,6 +953,7 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
 
         // # Global allocations
         if let Some(global_alloc) = self.tcx.try_get_global_alloc(id) {
+            // NOTE: static alignment from attributes has already been applied to the allocation.
             let (size, align) = global_alloc.size_and_align(*self.tcx, self.typing_env);
```

Comment:
The `GlobalAlloc::Static` case in `size_and_align` still needs to check the attribute I think.

I can't think of a good way to deduplicate that with `create_static_alloc`.